OpenRA.Mods.Common/Traits/World/MapOptions.cs

```diff
@@ -87,15 +87,15 @@ IEnumerable<LobbyOption> ILobbyOptions.LobbyOptions(Ruleset rules)
 			var gameSpeeds = Game.ModData.Manifest.Get<GameSpeeds>().Speeds
 				.ToDictionary(s => s.Key, s => s.Value.Name);
 
-			// NOTE: The server hardcodes special-case logic for this option id
+			// NOTE: This is just exposing the UI, the backend logic for this option is hardcoded in World
 			yield return new LobbyOption("gamespeed", GameSpeedDropdownLabel, GameSpeedDropdownDescription, GameSpeedDropdownVisible, GameSpeedDropdownDisplayOrder,
 				new ReadOnlyDictionary<string, string>(gameSpeeds), GameSpeed, GameSpeedDropdownLocked);
```

Comment:
The "Not Available" happens because we pass the null default of `GameSpeed` in as the initial state.

Tweaking the variable definitions above and passing `GameSpeed ?? gameSpeed.DefaultGameSpeed` like in the other places will fix it.